### PR TITLE
Add schemas to allow automatic validation of data files

### PR DESCRIPTION
**Pull request by hawkrives**
> This PR builds on top of #475.

Using [json-schema.org](http://json-schema.org) and [epoberezkin/ajv](https://github.com/epoberezkin/ajv), I created schema files for each data file in `data/`.

I also wrote a script to automatically validate the contents of the files against the schema, so we'll know quickly if we break something.

This setup actually caught an issue in the Tom Porter file that slipped by me during my data-splitting PR.

What to say about the schemas… They're actually written in YAML, not JSON, but I find YAML so much clearer than JSON to write and edit that I converted them over. (See https://gist.github.com/hawkrives/659d9e0af8f781533cba3e24cafb8370 for a comparison.) It's no more work to use YAML for the schema, since `ajv` runs in JavaScript-land anyway so it operates on JS objects in the first place.

I highly recommend https://spacetelescope.github.io/understanding-json-schema/basics.html for getting started with JSON-Schema, and http://camel.readthedocs.io/en/latest/yamlref.html for an overview of YAML.

I am willing to answer any questions about this PR, up to any including debates about if we even need schemas for the files (personally, I think we do.)

TL;DR: automatic validation of data in the data files. good? bad? silly?

### CODE REVIEW

**Review by drewvolz — APPROVED** [review]

**Review by elijahverdoorn — APPROVED** [review]
I like schemas in general, I don't see a reason not to use them for the data files.

**Comment by hawkrives** [comment]
Thanks all! I'll merge this after #475 is merged.

**Comment by hawkrives** [comment]
now, github, really?
